cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/AnalysisConfiguration.kt

```diff
@@ -285,7 +288,9 @@ class Forward(graphToFollow: GraphToFollow) : AnalysisDirection(graphToFollow) {
                         scope = scope,
                         sensitivities = sensitivities,
                     )
-                    .map { (edge, newCtx) -> this.unwrapNextStepFromEdge(edge) to newCtx }
+                    .map { (edge, newCtx) ->
+                        this.unwrapNextStepFromEdge(edge) to newCtx.addAssumptionDependence(edge)
```

Comment:
as this code repeats and is quite verbose, can we somehow simplify that or move it to a common function?

Reply:
I implemented an unwrapper function for this.